lib/kubeclient/common.rb

```diff
@@ -508,5 +553,11 @@ def http_options(uri)
 
       options.merge(@socket_options)
     end
+
+    def http_err(json_error_msg, e)
+      err_message = json_error_msg['message'] || e.message
+      error_klass = e.http_code == 404 ? ResourceNotFoundError : HttpError
```

Comment:
is there a way not to hardcode 404 here and in the row below?

Reply:
@abonas was talking about extracting 404 to a constant if I understand correctly. I'm okay with both ways.